NeoOnnx/src/Optimization/LayerNormFusionOptimizer.cpp

```diff
@@ -0,0 +1,360 @@
+{
+	if( skipBroadcast ) {
+		CBroadcastLayer* broadcastLayer = getExactLayer<CBroadcastLayer>( layer, /*addToSelection*/true );
+		if( broadcastLayer != nullptr && broadcastLayer->GetInputCount() == 1 ) {
+			// No additional checks
+			return broadcastLayer;
+		}
+	}
+	COnnxTransformHelper* transformHelperLayer = getExactLayer<COnnxTransformHelper>( layer, /*addToSelection*/true );
+	if( transformHelperLayer != nullptr && transformHelperLayer->GetInputCount() == 1 ) {
+		auto isEmptyRule = []( TBlobDim value, TBlobDim index ) -> bool {
+			return value == index || ( /* invalid-value */ value < 0 || value >= BD_Count );
+		};
+		const bool success =
+			( isEmptyRule( transformHelperLayer->GetRule( BD_BatchLength ), BD_BatchLength )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_BatchWidth ), BD_BatchWidth )
+			&& transformHelperLayer->GetRule( BD_ListSize ) == BD_Height
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Height ), BD_Height )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Width ), BD_Width )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Depth ), BD_Depth )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Channels ), BD_Channels ) )
+			||
+			( isEmptyRule( transformHelperLayer->GetRule( BD_BatchLength ), BD_BatchLength )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_BatchWidth ), BD_BatchWidth )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_ListSize ), BD_ListSize )
+			&& transformHelperLayer->GetRule( BD_Height ) == BD_ListSize
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Width ), BD_Width )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Depth ), BD_Depth )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Channels ), BD_Channels ) )
+			||
+			( transformHelperLayer->GetRule( BD_BatchLength ) == BD_ListSize
+			&& transformHelperLayer->GetRule( BD_BatchWidth ) == BD_Height
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_ListSize ), BD_ListSize )
+			&& transformHelperLayer->GetRule( BD_Height ) == BD_Channels
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Width ), BD_Width )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Depth ), BD_Depth )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Channels ), BD_Channels ) )
+			||
+			( isEmptyRule( transformHelperLayer->GetRule( BD_BatchLength ), BD_BatchLength )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_BatchWidth ), BD_BatchWidth )
+			&& transformHelperLayer->GetRule( BD_ListSize ) == BD_BatchLength
+			&& transformHelperLayer->GetRule( BD_Height ) == BD_BatchWidth
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Width ), BD_Width )
+			&& isEmptyRule( transformHelperLayer->GetRule( BD_Depth ), BD_Depth )
+			&& transformHelperLayer->GetRule( BD_Channels ) == BD_Height );
+		return success ? transformHelperLayer : nullptr;
+	}
+	CTransposeLayer* transposeLayer = getExactLayer<CTransposeLayer>( layer, /*addToSelection*/true );
```

Comment:
This code won't work. NeoOnnx import doesn't use `CTransformLayer` and `CTransposeLayer` anymore. It uses only onnx transpose/transform helpers.

Reply:
CTransformLayer & CTransposeLayer removed